pilot/pkg/xds/bootstrapds.go

```diff
@@ -54,9 +56,23 @@ func (e *BootstrapGenerator) Generate(proxy *model.Proxy, _ *model.PushContext,
 	if err = jsonpb.Unmarshal(io.Reader(&buf), bs); err != nil {
 		log.Warnf("failed to unmarshal bootstrap from JSON %q: %v", buf.String(), err)
 	}
+	bs = e.applyPatches(bs, proxy, push)
 	return model.Resources{
 		&discovery.Resource{
 			Resource: util.MessageToAny(bs),
 		},
 	}, model.DefaultXdsLogDetails, nil
 }
+
+func (e *BootstrapGenerator) applyPatches(bs *bootstrapv3.Bootstrap, proxy *model.Proxy, push *model.PushContext) *bootstrapv3.Bootstrap {
```

Comment:
I think we need safety guards here like https://github.com/istio/istio/blob/980b8f6bbeb205c7d471cd34b16b64ef6beb17cd/pilot/pkg/networking/core/v1alpha3/envoyfilter/rc_patch.go#L36-L39

Otherwise arbitrary inputs can crash Istio

Reply:
Done.